### DISCUSSION TITLE
✅ Port knocking question

### DISCUSSION THREAD

**Discussion by pwlnow** [isontheline/pro.webssh.net#558] [Q&A] [answered]
This port knocking config is to open a port on the destination server.  Is there a way to specify port knocking sequence to close the port?  If not, when will the port be closed again?  Will closing a connection automatically close the port in the firewall?

**Comment by isontheline** [comment]
Hello @pwlnow 👋

Port Knocking on WebSSH isn't related to open or close ports on your server. 

This task is managed by knockd daemon.

You can read about it here : https://help.ubuntu.com/community/PortKnocking

Do you need more information?

Have a nice day ☀️

**Comment by ghost** [comment]
I already understand the questioner. Knockd uses a sequence of ports to open the SSH port via firewall and another sequence to close the port via firewall. WebSSH seems to have a function to open the port by port knocking, but after the SSH session is finished, this port remains open, because WebSSH has no function to send the knocking sequence to close the SSH port again. So you would have to use a third tool to close the port again. I find this very inconvenient so I don't use this function.

**Reply by isontheline** [reply, marked as answer]
> So you would have to use a third tool to close the port again. I find this very inconvenient so I don't use this function.

I don't agree with you.

If you read this doc : https://help.ubuntu.com/community/PortKnocking

You will see that (example 2/3) the port is closed XX seconds after the knock open sequence.
